jaiarobotics/jaiabot
# bug/imu-reporting-data-but-not-changing/JAIA-1735

## Description

In the field we saw the bno085 imu stop reporting, but no error alerted the user. The imu never recovered from this. This code handles bno085 imu runtime error and reconnects to the imu.

## Error

![frozen-imu-data-journal-ctl](https://github.com/user-attachments/assets/04f38aa2-597d-45ed-8cf8-219f391e4683)

## Testing

* Tested on bot 2 fleet 0
* Open liaison and watch the imu data
* Start the reset line service for the bno085 (sudo systemctl start jaia_firm_bno085_reset_gpio_pin_py)
* The values should stop reporting for 2-3 seconds
* The values should start reporting again
